pkg/process/checks/process_linux.go

```diff
@@ -121,26 +122,40 @@ func mapWLMProcToProc(wlmProc *workloadmetacomp.Process, stats *procutil.Stats)
 		Uids:     wlmProc.Uids,
 		Gids:     wlmProc.Gids,
 		Stats:    stats,
-		Ports:    ports,
+		TCPPorts: tcpPorts,
+		UDPPorts: udpPorts,
 		Language: wlmProc.Language,
 		Service:  service,
 	}
 }
 
-// formatPorts converts a list of uin16 ports to a int32 PortInfo
-// TODO: because the service discovery response does not distinguish between tcp and udp currently, we currently send everything as TCP
-func formatPorts(ports []uint16) *model.PortInfo {
-	// if ports were not collected, we want to semantically indicate that no data was collected instead of
+// formatPorts converts separate TCP and UDP uint16 port lists to a int32 PortInfo
+func formatPorts(tcpPorts, udpPorts []uint16) *model.PortInfo {
+	// if both port lists are nil, we want to semantically indicate that no data was collected instead of
 	// returning no open ports
-	if ports == nil {
+	if tcpPorts == nil && udpPorts == nil {
```

Comment:
Just to make sure I understand correctly, if we collected only TCP ports, then the UDP ports slice will be `nil` and not `[]int32{}`.